python/ray/air/util/tensor_extensions/utils.py

```diff
@@ -1,7 +1,11 @@
-from typing import Any
+from typing import Any, Sequence, Union, TYPE_CHECKING
+import warnings
 
 import numpy as np
 
+if TYPE_CHECKING:
```

Comment:
This is the new section added in this PR. The rest of the code changes are exactly the same as from the [previous PR](https://github.com/ray-project/ray/pull/31544)